drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -577,7 +606,16 @@ static int nrf5_start(const struct device *dev)
 
 static int nrf5_stop(const struct device *dev)
 {
+#if defined(CONFIG_OPENTHREAD_CSL_RECEIVER)
+	if (nrf_802154_sleep_if_idle() != NRF_802154_SLEEP_ERROR_NONE) {
+		if (nrf5_data.event_handler) {
```

Comment:
What would happen if there is no `event_handler` set? I think `else` block should be somehow implemented.